### PR TITLE
[CFM-755] send wellcome mail for users created by admin

### PR DESCRIPTION
**Pull request by anvmn**
https://issuetracker.amplexor.com/jira/browse/CFM-755?filter=-1

### CODE REVIEW

**Comment by anvmn** [comment]
selection_097
selection_097

**Comment by anvmn** [comment]
@ordavidil 
Camparing recieved mails with notification documentation (https://wiki.amplexor.com/confluence/display/C4DNG/FF+Mail+templates), I see that we don't get exectly what's documented (MT1, MT3).

'From' field is incorrect.
Email template at some place is not per requirements.
Email footer (--- capacity4more team) is incorrect.

Should I edit the templates per documentation?

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.module [hunk 1]
```diff
@@ -569,6 +569,14 @@ function c4m_user_profile_form_user_register_form_submit($form, &$form_state) {
     )
   );
 
+  // Always send "welcome" mail for user created by admin.
+  $admin = user_access('administer users');
+  if ($admin) {
+    $account = user_load($form['#user']->uid);
+    $params['account'] = $account;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.module]: `$params` seems unused.

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.module [hunk 2]
```diff
@@ -569,6 +569,14 @@ function c4m_user_profile_form_user_register_form_submit($form, &$form_state) {
     )
   );
 
+  // Always send "welcome" mail for user created by admin.
+  $admin = user_access('administer users');
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.module]: Remove this line.

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.module [hunk 3]
```diff
@@ -569,6 +569,14 @@ function c4m_user_profile_form_user_register_form_submit($form, &$form_state) {
...
+  if ($admin) {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.module]:
```php
if (user_access('administer users')) {
```

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.module [hunk 4]
```diff
@@ -569,6 +569,14 @@ function c4m_user_profile_form_user_register_form_submit($form, &$form_state) {
...
+    $account = user_load($form['#user']->uid);
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.module]: @anvmn 
Did you check if `$form['#user']` is already the object of the user ?
we might not need the `user_load()` in here.

**Comment by ordavidil** [comment]
@anvmn 
Do the user gets more than one email after your PR?
Did the user get an email about this action __before__ your PR?
if not than maybe it will be easier to create a message and use `message_notify`, I believe it will be much more easy to handle the template ...

**Comment by anvmn** [comment]
@ordavidil 
Before PR:
In case 'Notify user of new account' on create user form is checked, user gets MT1 (new user created by administrator) notification. If not, user gets no mails.
This functionality is not changed by PR.
After PR:
Every user created by admin gets MT3 'welcome' notification - same mail user gets after manually activating his account (so yes, this template was in use before).
If 'Notify user of new account' is checked, user gets additional mail (MT1). In total, 2 mails - MT1 and MT3.
If 'Notify user of new account' is not checked, only MT3.

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.module (hunk 4 above)
Comment (anvmn) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.module]: @ordavidil 
Indeed, object of the user.

**Comment by anvmn** [comment]
@ordavidil 
What should I do with templates inaccuracies?

**Pull request merged**